flow/src/main/java/com/sequenceiq/flow/rotation/handler/ExecuteRotationHandler.java

```diff
@@ -33,7 +39,12 @@ protected Selectable defaultFailureEvent(Long resourceId, Exception e, Event<Exe
 
     @Override
     protected Selectable doAccept(HandlerEvent<ExecuteRotationTriggerEvent> event) {
-        secretRotationService.executeRotation(event.getData().getSecretType(), event.getData().getResourceCrn(), event.getData().getExecutionType());
-        return ExecuteRotationFinishedEvent.fromPayload(event.getData());
+        ExecuteRotationTriggerEvent rotationEvent = event.getData();
+        if (rotationEvent.getExecutionType() == null) {
```

Comment:
null check should be moved into SecretRotationUsageProcessor by adding executionType as a new parameter

Reply:
you are right, fixed